### ISSUE TITLE
Incorrect cmorisation of Ofx deptho

### ISSUE THREAD

**Issue by treerink** [EC-Earth/ece2cmor3#530]
The quality control of `Ofx deptho` shows that its cmorisation is incorrect.

So what is the matter with `Ofx deptho'?

Looking into the `Bathymetry` `(=deptho)` data, one sees zero values at the continents, zero is of course usually not the best invalid value, anyway I would consider this as a minor omission? But what else is the matter:

According to the data request this should be reported as missing over land grid cells. I agree that at first sight it seems like not a huge problem if land is instead reported as zero depth. But actually this is not what is happening: If we take land grid cells to be those where the
ocean model is not operating, the largest problematic area is the caspian sea, the largest body of inland water that is completely missing from the model output, but reported with depth between 15m and 45m in `deptho.` We really should use the (fixed) `sftof` to apply the correct mask.

**Comment by goord** [comment]
Ok so we should explicitly mask it with sftof? Where does that variable come from, is it in the nemo output data?

**Comment by goord** [comment]
To answer my own question: yes it is...
@treerink can you send me some output + your bathy file so I can test?

**Comment by treerink** [comment]
Hi Gijs,

At the bull, the cmorised sftof (data-qa corrected):
```
/lustre3/projects/CMIP6/reerink/cmorised-results/cmor-cmip-historical/h003/CMIP6/CMIP/EC-Earth-Consortium/EC-Earth3/historical/r1i1p1f1/part-2/Ofx/sftof/gn/v20190926/sftof_Ofx_EC-Earth3_historical_r1i1p1f1_gn.nc
```
and
```
/lustre3/projects/CMIP6/reerink/ec-earth-3/branch-r6485-historical-setup/h003/output/nemo/ofx-data/bathy_meter.nc
```

**Comment by treerink** [comment]
Also here:
```
/net/bhw463/nobackup/users/reerink/voor-gijs/
```

**Comment by goord** [comment]
This should be fixed in the deptho-mask-fix branch, please test

**Comment by treerink** [comment]
Test runs successful using:
```
ece2cmor3/ece2cmor3/resources/test-data-request/varlist-nemo-Ofx-deptho.json
```

**Comment by treerink** [comment]
Anyone willing to test `deptho` with ESMval? @zklaus?

It can be tested from the master branch now.

**Comment by zklaus** [comment]
We'll have a look here at SMHI. Might take a couple of days, though.

@goord, @treerink cheers for the fix though!

**Comment by treerink** [comment]
Having heard no complains, closing this issue. Please reopen if necessary.

**Comment by zklaus** [comment]
Sorry, the data looks good, but something strange is going on with the coordinates. Both latitudes and longitudes have parts over Antarctica set to 0.

**Comment by treerink** [comment]
Indeed they are easy to spot in the netcdf file, the same pattern of zeros in the `latitudes` as in the `longitudes.` It concerns only points which have an invalid value in `deptho`.

**Comment by zklaus** [comment]
What do you mean by
> invalid value in `deptho`

? It is land points, so they are masked, by they should have correct coordinates still, right?

**Comment by treerink** [comment]
Also, the same pattern is also visible in the `vertices_latitude` and `vertices_longitude`, though the latter has an additional feature (probably at its cyclic boundary).

**Comment by treerink** [comment]
Yes they should have proper coordinates I would say. I just notice that all `deptho` values with non-invalid values, have well defined coordinates.

**Comment by treerink** [comment]
I had a look in the raw data file which is used for `deptho:` `output/nemo/ofx-data/bathy_meter.nc` and exactly the same weird zero value patterns are visible in `nav_lat` `nav_lon` in that file.

So it seems this values enter already from the raw data.

**Comment by goord** [comment]
I can try to use the T-grid from regular output files for this instead of the (apparently corrupt?) bathy_meter file...

**Comment by goord** [comment]
Implemented fix in bathy-grid-fix branch, ready for testing

**Comment by treerink** [comment]
Tested and looks fine now. The `deptho` field itself remained identical.

**Comment by treerink** [comment]
Solved with merging #552.

Closing.
